app/views/issued_certificate_mailer/issued.html.erb

```diff
@@ -7,21 +7,20 @@
 <% content_for(:title_button_link, link) %>
 
 <% content_for :salutation do %>
-  Hello <%= user.name %>,
+  <%= t("shared.salutation_name", name: user.name) %>
 <% end %>
 
 <% content_for :body do %>
   <p>
-    <b>Congratulations!</b>
+    <b><%= t("views.mailers.issued.congratulations") %></b>
   </p>
 
   <p>
-    You've been awarded a certificate for completing the <strong><%= course.name %></strong> course.
+    <%= t("views.mailers.issued.awarded_certificate", course_name: course.name).html_safe %>
```

Comment:
@pupilfirst/engineers I'm a bit concerned about the non-standard way we're handling interpolation here. I think to keep things safe, the _Rails-way_ is to have these keys that contain HTML be suffixed with `_html`, and  then to avoid using the `.html_safe` in the translation calls.

Here are the docs with examples: https://guides.rubyonrails.org/i18n.html#using-safe-html-translations

Using translated text that includes interpolation in this way should result in escaping only of the user-supplied values in the resulting text. That's what the docs say.

However, while this may work for Rails-rendered views, this approach won't work for client-side translations where such text would be injected into the DOM using `dangerouslySetInnerHTML`. I did a bit of looking up, and we may need to implement similar logic in the `I18n` module using DOMPurify. i.e., if the supplied key ends with `_html`, and if there are variables supplied for to the translation call, use DOMPurify to clean all variables before using it in the translation.

Here's a post in the `i18n-js` repo that includes an old implementation of this idea: https://github.com/fnando/i18n-js/issues/310#issuecomment-305909522

Reply:
Thanks, I will update PR on the next week